src/functional_pipeline/functional_pipeline.m

```diff
@@ -62,20 +65,63 @@
 
 
 % PARSE INPUT
-[subjectDir, varargin{:}] = convertStringsToChars(subjectDir, varargin{:});
+[subjectSelection, varargin{:}] = convertStringsToChars(subjectSelection, varargin{:});
 
-assert(ischar(subjectDir), ...
+assert(ischar(subjectSelection), ...
```

Comment:
I would update `subjectDirNotText` to `subjectSelectionNotText` (here and at the other locations), such that the internal error ID matches the variable.